clients/vault/src/oracle/agent.rs

```diff
@@ -57,82 +84,76 @@ async fn handle_message(
-/// Returns an `OracleAgent` that will handle incoming messages from Stellar Node,
-/// and to send messages to Stellar Node
-pub async fn start_oracle_agent(
+pub async fn listen_for_stellar_messages(
 	config: StellarOverlayConfig,
-	secret_key: &str,
+	collector: Arc<RwLock<ScpMessageCollector>>,
+	secret_key_as_string: String,
 	shutdown_sender: ShutdownSender,
-) -> Result<OracleAgent, Error> {
-	let is_public_network = config.is_public_network();
+) -> Result<(),service::Error<crate::Error>> {
+	tracing::info!("listen_for_stellar_messages(): Starting connection to Stellar overlay network...");
 
-	tracing::info!("start_oracle_agent(): Starting connection to Stellar overlay network...");
+	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key_as_string)
+		.await.map_err(|e|{
+			tracing::error!("listen_for_stellar_messages(): Failed to connect to Stellar overlay network: {e:?}");
+			service::Error::StartOracleAgentError
+		})?;
 
-	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key).await?;
 	// use StellarOverlayConnection's sender to send message to Stellar
 	let sender = overlay_conn.sender();
 
-	let collector = Arc::new(RwLock::new(ScpMessageCollector::new(
-		config.is_public_network(),
-		config.stellar_history_archive_urls(),
-	)));
-	let collector_clone = collector.clone();
-
-	let shutdown_sender_clone = shutdown_sender.clone();
-	// disconnect signal sender tells the StellarOverlayConnection to close its TcpStream to Stellar
-	// Node
-	let (disconnect_signal_sender, mut disconnect_signal_receiver) = mpsc::channel::<()>(2);
-
-	let sender_clone = overlay_conn.sender();
-	tokio::spawn(async move {
-		loop {
-			tokio::select! {
-				_ = sleep(Duration::from_millis(100)) => {
-					tracing::info!("start_oracle_agent(): go to sleep");
-				},
-				// if a disconnect signal was sent, disconnect from Stellar.
-				result = disconnect_signal_receiver.recv() => {
-					if result.is_none() {
-						tracing::info!("start_oracle_agent(): disconnect overlay...");
-						break
-					}
-				},
-				result = overlay_conn.listen() => {
-					tracing::info!("start_oracle_agent(): received message from overlay");
-					match result {
-					Ok(Some(msg)) => {
-						let msg_as_str = to_base64_xdr_string(&msg);
-						if let Err(e) = handle_message(msg, collector_clone.clone(), &sender_clone).await {
-							tracing::error!("start_oracle_agent(): failed to handle message: {msg_as_str}: {e:?}");
-						}
-					},
-					Ok(None) => {},
-					// connection got lost
-					Err(e) => {
-						tracing::error!("start_oracle_agent(): encounter error in overlay: {e:?}");
-
-						if let Err(e) = shutdown_sender_clone.send(()) {
-							tracing::error!("start_oracle_agent(): Failed to send shutdown signal in thread: {e:?}");
-						}
-						break
-					},
-				}},
+	// log a new message received, every 1 minute.
+	let interval = Duration::from_secs(60);
```

Comment:
This is intentional, to keep logging every minute to show whether the vault is stuck or not.